selfservice/strategy/oidc/provider_config.go

```diff
@@ -116,61 +117,41 @@ type ConfigurationCollection struct {
 	Providers       []Configuration `json:"providers"`
 }
 
+// !!! WARNING !!!
+//
+// If you add a provider here, please also add a test to
+// provider_private_net_test.go
+var supportedProviders = map[string]func(config *Configuration, reg dependencies) Provider{
```

Comment:
This refactoring is needed to be able to mock a new provider in that can selectively verify or fail verification of an ID Token.

I've double-checked, that I migrated all providers, but *please* check this again! Thank you.